Contributor Documentation/Debugging Memory Leaks.md

```diff
@@ -0,0 +1,19 @@
+# Debugging Memory Leaks
```

Comment:
Could add a link to eg. https://www.swift.org/documentation/server/guides/memory-leaks-and-usage.html with this being a TLDR?

Reply:
Updated